### PR TITLE
Added TS compatibility + tests

### PR DESCRIPTION
**Pull request by andreamalvetta**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -55,8 +55,8 @@ function jsonImporter(options = {}) {
   };
 }
 
-function isJSONfile(url) {
-  return /\.js(on5?)?$/.test(url);
+function isValidFile(url) {
+  return /\.[tj]s(on5?)?$/.test(url);
```
Comment (DanMad) [inline src/index.js:59]: This regex would return true if the file extension was `.tson` and `.tson5`. Suggesting /\.(ts|js(on5?)?)$/ instead...

Comment (andreamalvetta) [reply src/index.js:59]: Good catch dude

**Review by DanMad — CHANGES_REQUESTED** [review]
Just the one thing I noticed with the regex in `isValidFile()`...

**Review by DanMad — APPROVED** [review]
Nice!

File: src/index.js (hunk 1 above)
Comment (andreamalvetta) [reply src/index.js]: Done

**Pull request merged**